R/S3-methods.R

```diff
@@ -37,13 +37,15 @@ print.sjmisc_frq <- function(x, ...) {
     insight::print_color(sprintf(
       "# total N=%i  valid N=%i  mean=%.2f  sd=%.2f\n\n",
       sum(dat$frq, na.rm = TRUE),
-      sum(dat$frq[1:(nrow(dat) - 1)], na.rm = TRUE),
+      sum(dat$frq[0:(nrow(dat) - 1)], na.rm = TRUE),

```

Comment:
Can you give me a hint why you start from `0` here?

Reply:
Yes, sure. To allow to show the frequency of a vector with only NA values, the valid N printed should be 0. Before this change, since in that case the data frame of frequencies consists of only 1 row of missing values, valid N printed was wrongly the length of the vector. Otherwise, as row 0 is always missing (I believe), count of valid frequencies can start by row 0.

Reply:
Still not sure I get the point... Is there a difference? Or do you have a small example that demonstrate the effect of your change? E.g.

``` r
x <- c(1,2,43)
x[0:3]
#> [1]  1  2 43
sum(x[0:3])
#> [1] 46
sum(x[1:3])
#> [1] 46
x[0]
#> numeric(0)
```

It doesn't matter if we start with 0 or 1, since `0` is no valid index.

Reply:
In general, it doesn't matter, but in the next case:
```
> library(sjmisc)
> x <- rep(NA,3)
> frq(x)

x <lgl>
# total N=3  valid N=0  mean=NaN  sd=NA

  val frq raw.prc valid.prc cum.prc
 <NA>   3     100        NA      NA
```
Since the data frame of frequencies consists of the row `<NA>   3     100        NA      NA` only, starting from 1 the sum counts row 1 so the printed output would be
```
x <lgl>
# total N=3  valid N=3  mean=NaN  sd=NA

  val frq raw.prc valid.prc cum.prc
 <NA>   3     100        NA      NA
```
The difference is the `valid N=0` or `valid N=3`.